Plugin/CppApi/source/CoordinateConversionController.cpp

```diff
@@ -407,6 +503,99 @@ void CoordinateConversionController::clearResults()
     m_results->clearResults();
 }
 
+void CoordinateConversionController::addCoordinateFormat(const QString& newFormat)
+{
+  if (!m_coordinateFormats.contains(newFormat))
+    return;
+
+  auto it = m_options.cbegin();
+  auto itEnd = m_options.cend();
+  for (; it != itEnd; ++it)
+  {
+    const auto& option = *it;
+    if (option->name().compare(newFormat) == 0)
+      return;
+  }
+
+  CoordinateConversionOptions* option = new CoordinateConversionOptions(this);
+  option->setName(newFormat);
+
+  if (newFormat == DEGREES_DECIMAL_MINUTES_FORMAT)
+  {
+    option->setOutputMode(CoordinateConversionOptions::CoordinateType::CoordinateTypeLatLon);
+    option->setLatLonFormat(CoordinateConversionOptions::LatitudeLongitudeFormat::LatitudeLongitudeFormatDegreesDecimalMinutes);
+  }
+  else if (newFormat == USNG_FORMAT)
+  {
+    option->setOutputMode(CoordinateConversionOptions::CoordinateType::CoordinateTypeUsng);
+    option->setPrecision(7);
+    option->setAddSpaces(true);
+  }
+  else if (newFormat == UTM_FORMAT)
+  {
+    option->setOutputMode(CoordinateConversionOptions::CoordinateType::CoordinateTypeUtm);
+    option->setUtmConversionMode(CoordinateConversionOptions::UtmConversionMode::UtmConversionModeNorthSouthIndicators);
+    option->setAddSpaces(true);
+  }
+  else if (newFormat == DEGREES_MINUTES_SECONDS_FORMAT)
+  {
+    option->setOutputMode(CoordinateConversionOptions::CoordinateType::CoordinateTypeLatLon);
+    option->setLatLonFormat(CoordinateConversionOptions::LatitudeLongitudeFormat::LatitudeLongitudeFormatDegreesMinutesSeconds);
+    option->setDecimalPlaces(6);
+  }
+  else if (newFormat == MGRS_FORMAT)
+  {
+    option->setOutputMode(CoordinateConversionOptions::CoordinateType::CoordinateTypeMgrs);
+    option->setMgrsConversionMode(CoordinateConversionOptions::MgrsConversionMode::MgrsConversionModeAutomatic);
+  }
+  else if (newFormat == DECIMAL_DEGREES_FORMAT)
+  {
+    option->setOutputMode(CoordinateConversionOptions::CoordinateType::CoordinateTypeLatLon);
+    option->setLatLonFormat(CoordinateConversionOptions::LatitudeLongitudeFormat::LatitudeLongitudeFormatDecimalDegrees);
+  }
```

Comment:
Should these all be stored somewhere as the format defaults? It might be easier for users to locate/customize that way.